tests/python_package_test/test_sklearn.py

```diff
@@ -279,3 +279,13 @@ def test_predict(self):
         self.assertRaises(AssertionError,
                           np.testing.assert_allclose,
                           res_engine, res_sklearn_params)
+
+    def test_evaluate_train_set(self):
+        X, y = load_boston(True)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+        gbm = lgb.LGBMRegressor(n_estimators=10, silent=True)
+        gbm.fit(X_train, y_train, eval_set=[(X_train, y_train), (X_test, y_test)], verbose=False)
+        self.assertTrue('training' in gbm.evals_result_)
```

Comment:
There is `assertIn` check: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn
It is already used in our tests:
https://github.com/Microsoft/LightGBM/blob/cba824474897c8d7e71a3df261faaefa091a40c5/tests/python_package_test/test_engine.py#L383